WordPress/Classes/Networking/MenusServiceRemote.m

```diff
@@ -0,0 +1,379 @@
+#import "MenusServiceRemote.h"
+#import "Blog.h"
+#import "WordPressComApi.h"
+#import "RemoteMenu.h"
+#import "RemoteMenuItem.h"
+#import "RemoteMenuLocation.h"
+
+NSString * const MenusRemoteKeyID = @"id";
+NSString * const MenusRemoteKeyMenu = @"menu";
+NSString * const MenusRemoteKeyMenus = @"menus";
+NSString * const MenusRemoteKeyLocations = @"locations";
+NSString * const MenusRemoteKeyContentID = @"content_id";
+NSString * const MenusRemoteKeyDescription = @"description";
+NSString * const MenusRemoteKeyLinkTarget = @"link_target";
+NSString * const MenusRemoteKeyLinkTitle = @"link_title";
+NSString * const MenusRemoteKeyName = @"name";
+NSString * const MenusRemoteKeyType = @"type";
+NSString * const MenusRemoteKeyTypeFamily = @"type_family";
+NSString * const MenusRemoteKeyTypeLabel = @"type_label";
+NSString * const MenusRemoteKeyURL = @"url";
+NSString * const MenusRemoteKeyItems = @"items";
+NSString * const MenusRemoteKeyDeleted = @"deleted";
+NSString * const MenusRemoteKeyLocationDefaultState = @"defaultState";
+
+@implementation MenusServiceRemote
+
+#pragma mark - Remote queries: Creating and modifying menus
+
+- (void)createMenuWithName:(NSString *)menuName
+                      blog:(Blog *)blog
+                   success:(MenusServiceRemoteMenuRequestSuccessBlock)success
+                   failure:(MenusServiceRemoteFailureBlock)failure
+{
+    NSNumber *blogId = [blog dotComID];
+    NSParameterAssert([blogId isKindOfClass:[NSNumber class]]);
+    NSParameterAssert([menuName isKindOfClass:[NSString class]]);
+    
+    NSString *path = [NSString stringWithFormat:@"sites/%@/menus/new", blogId];
+    NSString *requestURL = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
+    
+    [self.api POST:requestURL
+        parameters:@{MenusRemoteKeyName: menuName}
+           success:^(AFHTTPRequestOperation * _Nonnull operation, id  _Nonnull responseObject) {
+               if (success) {
+                   NSAssert([responseObject isKindOfClass:[NSDictionary class]], @"Expected a dictionary");
```

Comment:
Love it. Pushed a commit revising the error handling of bad responses.

Note: We would keep the assertions for parsing the data on valid responses as those should throw for debugging the returned data model.